modules/drouting/dr_load.c

```diff
@@ -641,7 +641,7 @@ rt_data_t* dr_load_routing_info(struct head_db *part,
 			for(i=0; i < RES_ROW_N(res); i++) {
 				row = RES_ROWS(res) + i;
 				/* RULE_ID column */
-				check_val( rule_id_drr_col, ROW_VALUES(row), DB_INT, 1, 0);
+				check_val2( rule_id_drr_col, ROW_VALUES(row), DB_INT, DB_BIGINT, 1, 0);
 				int_vals[INT_VALS_RULE_ID_DRR_COL] = VAL_INT (ROW_VALUES(row));
```

Comment:
similar, read as VAL_INT or VAL_BIGINT.

Also, the rule ID is exposed as AVP to script, so as an integer :-/ (for Carriers and GWs we have string ID which is used only for log printing and it is not stored into internal structures), so it may be subject to an overflow (while reading 64b from DB)

So, at least, we need to upgrade the rule->id to long long (and also check the implications of that - not sure where this ID is used further in the code). And the `rule_id_avp` to expose the ID as string, not int.